pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -634,154 +635,159 @@ def getData(self):
+            y = np.diff(self.yData)/np.diff(self.xData)
+            x = x[:-1]
+        if self.opts['phasemapMode']:  # plot dV/dt vs V
+            x = self.yData[:-1]
+            y = np.diff(self.yData)/np.diff(self.xData)
+                
+        with np.errstate(divide='ignore'):
+            if self.opts['logMode'][0]:
+                x = np.log10(x)
+            if self.opts['logMode'][1]:
+                if np.issubdtype(y.dtype, np.floating):
+                    eps = np.finfo(y.dtype).eps
+                else:
+                    eps = 1
+                y = np.sign(y) * np.log10(np.abs(y)+eps)
+
+        ds = self.opts['downsample']
+        if not isinstance(ds, int):
+            ds = 1
+
+        if self.opts['autoDownsample']:
+            # this option presumes that x-values have uniform spacing
+            if v_rng is not None and len(x) > 1:
+                dx = float(x[-1]-x[0]) / (len(x)-1)
+                if dx != 0.0:
+                    x0 = (v_rng.left()-x[0]) / dx
+                    x1 = (v_rng.right()-x[0]) / dx
+                    width = self.getViewBox().width()
+                    if width != 0.0:
+                        ds = int(max(1, int((x1-x0) / (width*self.opts['autoDownsampleFactor']))))
+                    ## downsampling is expensive; delay until after clipping.
+
+        if self.opts['clipToView']:
+            if view is None or view.autoRangeEnabled()[0]:
+                pass # no ViewBox to clip to, or view will autoscale to data range.
+            else:
+                # clip-to-view always presumes that x-values are in increasing order
+                if v_rng is not None and len(x) > 1:
+                    # print('search:', v_rng.left(),'-',v_rng.right() )
+                    # find first in-view value (left edge) and first out-of-view value (right edge)
+                    # since we want the curve to go to the edge of the screen, we need to preserve
+                    # one down-sampled point on the left and one of the right, so we extend the interval
+                    x0 = np.searchsorted(x, v_rng.left()) - ds
+                    x0 = fn.clip_scalar(x0, 0, len(x)) # workaround
+                    # x0 = np.clip(x0, 0, len(x))
+
+                    x1 = np.searchsorted(x, v_rng.right()) + ds
+                    x1 = fn.clip_scalar(x1, 0, len(x))
+                    # x1 = np.clip(x1, 0, len(x))
+                    x = x[x0:x1]
+                    y = y[x0:x1]
+
+        if ds > 1:
+            if self.opts['downsampleMethod'] == 'subsample':
+                x = x[::ds]
+                y = y[::ds]
+            elif self.opts['downsampleMethod'] == 'mean':
+                n = len(x) // ds
+                # x = x[:n*ds:ds]
+                stx = ds//2 # start of x-values; try to select a somewhat centered point
+                x = x[stx:stx+n*ds:ds] 
+                y = y[:n*ds].reshape(n,ds).mean(axis=1)
```

Comment:
the starting offset of ds//2  (e.g. 2 for ds=5) picks a somewhat centered x value to plot the mean y value over.